tests/Test.ADAL.NET.Unit/JsonWebTokenTests.cs

```diff
@@ -51,6 +52,54 @@ public class JsonWebTokenTests
     {
         private PlatformParameters platformParameters;
 
+        MockHttpMessageHandler X5CMockHandler = new MockHttpMessageHandler()
+        {
+            Method = HttpMethod.Post,
+            ResponseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent("{\"token_type\":\"Bearer\",\"expires_in\":\"3599\",\"access_token\":\"some-access-token\"}")
+            },
+            AdditionalRequestValidation = request =>
+            {
+                var requestContent = request.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                var formsData = EncodingHelper.ParseKeyValueList(requestContent, '&', true, null);
+
+                // Check presence of client_assertion in request
+                string encodedJwt;
+                Assert.IsTrue(formsData.TryGetValue("client_assertion", out encodedJwt), "Missing client_assertion from request");
+
+                // Check presence of x5c cert claim. It should exist.
+                JwtSecurityTokenHandler handler = new JwtSecurityTokenHandler();
+                var jsonToken = handler.ReadToken(encodedJwt) as JwtSecurityToken;
+                var x5c = jsonToken.Header.Where(header => header.Key == "x5c").FirstOrDefault();
+                Assert.IsTrue(x5c.Key == "x5c");
+            }
+        };
+
+        MockHttpMessageHandler EmptyX5CMockHandler = new MockHttpMessageHandler()
+        {
+            Method = HttpMethod.Post,
+            ResponseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent("{\"token_type\":\"Bearer\",\"expires_in\":\"3599\",\"access_token\":\"some-access-token\"}")
+            },
+            AdditionalRequestValidation = request =>
+            {
+                var requestContent = request.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                var formsData = EncodingHelper.ParseKeyValueList(requestContent, '&', true, null);
+
+                // Check presence of client_assertion in request
+                string encodedJwt;
+                Assert.IsTrue(formsData.TryGetValue("client_assertion", out encodedJwt), "Missing client_assertion from request");
+
+                // Check presence of x5c cert claim. It should not exist.
+                JwtSecurityTokenHandler handler = new JwtSecurityTokenHandler();
+                var jsonToken = handler.ReadToken(encodedJwt) as JwtSecurityToken;
```

Comment:
You can use: JwtSecurityTokenHandler.ReadJwtToken(...) and you don't have to cast as it returns as JwtSecurityToken.